src/test/java/net/snowflake/client/jdbc/ResultSetIT.java

```diff
@@ -1143,6 +1124,64 @@ public void testMultipleChunks() throws SQLException, IOException {
     telemetry.sendBatchAsync();
   }
 
+  @Test
+  public void testMetadataAPIMetricCollection() throws SQLException {
+    Connection con = getConnection();
+    Telemetry telemetry =
+        con.unwrap(SnowflakeConnectionV1.class).getSfSession().getTelemetryClient();
+    DatabaseMetaData metadata = con.getMetaData();
+    // Call one of the DatabaseMetadata API functions but for simplicity, ensure returned ResultSet
+    // is empty
+    metadata.getColumns("fakecatalog", "fakeschema", null, null);
+    LinkedList<TelemetryData> logs = ((TelemetryClient) telemetry).logBuffer();
+    // No result set has been downloaded from server so no chunk downloader metrics have been
+    // collected
+    // Logs should contain 1 item: the data about the getColumns() parameters
+    assertEquals(logs.size(), 1);
+    // Assert the log is of type client_metadata_api_metrics
+    assertEquals(
+        logs.get(0).getMessage().get(TelemetryUtil.TYPE).textValue(),
+        TelemetryField.METADATA_METRICS.toString());
+    // Assert function name and params match and that query id exists
+    assertEquals(logs.get(0).getMessage().get("function_name").textValue(), "getColumns");
+    assertTrue(
+        Pattern.matches(
+            "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}",
+            logs.get(0).getMessage().get("query_id").textValue()));
+    JsonNode parameterValues = logs.get(0).getMessage().get("function_parameters");
+    assertEquals(parameterValues.get("catalog").textValue(), "fakecatalog");
+    assertEquals(parameterValues.get("schema").textValue(), "fakeschema");
+    assertEquals(parameterValues.get("general_name_pattern").textValue(), null);
+    assertEquals(parameterValues.get("specific_name_pattern").textValue(), null);
+
+    // send data to clear log for next test
+    telemetry.sendBatchAsync();
+
+    metadata.getProcedureColumns("TESTDB", "TESTSCHEMA", null, null);
```

Comment:
Added a test case here for when a ResultSet is actually returned from the function call.